ui/js/directives/clonejobs.js

```diff
@@ -81,21 +81,22 @@ treeherder.directive('thCloneJobs', [
             };
         }
 
-        jobs = $(selector);
-        var idx = jobs.index(jobs.filter(".selected-job"));
-        idx = getIndex(idx, jobs);
+        jobs = $(selector).filter(":visible");
+        if (jobs.length) {
+            var selIdx = jobs.index(jobs.filter(".selected-job"));
+            var idx = getIndex(selIdx, jobs);
 
-        el = $(jobs[idx]);
-        while (el.css('display') === 'none') {
-            idx = getIndex(idx, jobs);
             el = $(jobs[idx]);
+            key = el.attr(jobKeyAttr);
+            if (jobMap && jobMap[key] && selIdx !== idx) {
+                selectJob(jobMap[key].job_obj);
+                return;
+            }
         }
-
-        key = el.attr(jobKeyAttr);
-        if (jobMap && jobMap[key]) {
-            selectJob(jobMap[key].job_obj);
-        }
-
+        // if there was no new job selected, then ensure that we clear any job that
+        // was previously selected.
+        $timeout($rootScope.closeJob, 0);
```

Comment:
Hm, I'm not so sure about closing the job details panel here. If I've selected a successful job in a view that is filtered, and I press the n,p key, my panel disappears. I would expect (I think?) just the thNotify to appear telling me there's no available unclassified failures, and my current job to remain selected + panel open.

eg. this scenario and I press n,p
![scenario](https://cloud.githubusercontent.com/assets/3660661/8561130/618a9caa-24ee-11e5-8779-4b590258326a.jpg)

Results in this:
![scenarioresult](https://cloud.githubusercontent.com/assets/3660661/8561147/9fc94840-24ee-11e5-9dda-5d3fd13576b7.jpg)